doc/read-the-docs-site/adr/0005-inlining-saturated-functions.md

```diff
@@ -0,0 +1,185 @@
+# ADR 5: Inlining fully saturated functions
+
+Date: 2022-12
+
+## Authors
+
+Marty Stumpf <marty.stumpf@iohk.io>
+
+## Status
+
+Accepted
+
+## Context
+
+As part of our effort to increase script capacity,
+we take from [Plutonomy](https://github.com/well-typed/plutonomy),
+which gives 15-30% speedup on benchmarks.
+See [PLT-1044](https://input-output.atlassian.net/browse/PLT-1044).
+
+The goal is to integrate what we know already works there and
+also add optimizations we can do before UPLC (Plutonomy only applies optimization to UPLC).
+We currently have an inliner pass for PIR and UPLC.
+One of the optimization we can add is inlining of saturated functions in both PIR and UPLC.
+From our analysis, this is one of the optimizations that contributes to the speedup in Plutonomy.
```

Comment:
Is this the totality of our justification? We discussed that it wasn't totally clear _why_ inlining saturated functions is good, did you have more thoughts on that? which also bears on what we consider to be saturated